command/agent/config.go

```diff
@@ -318,6 +318,12 @@ type ClientConfig struct {
 	// doest not exist Nomad will attempt to create it during startup. Defaults to '/nomad'
 	CgroupParent string `hcl:"cgroup_parent"`
 
+	// NativeServiceDiscovery is a boolean parameter which allows operators to
+	// enable/disable to Nomad native service discovery feature on the client.
+	// This parameter is exposed via the Nomad fingerprinter and used to ensure
+	// correct scheduling decisions on allocations which require this.
+	NativeServiceDiscovery *bool `hcl:"native_service_discovery"`
```

Comment:
Ok I guess this is our last chance to bikeshed so I'll bite:

Should we call this this `NomadServiceDiscovery` / `nomad_service_discovery`?

I know having a self-referential configuration name is weird, but if we ever added an option to disable the Consul provider it would have slightly nicer symmetry: `{nomad,consul}_service_discovery = {true,false}`

Feel free to ignore... just thought I'd put thoughts down in case anybody has strong feelings one way or the other.

Reply:
I think this makes sense and lines up more with the fingerprint value. The self reference is a little odd, but in the larger context as you mention it sounds reasonable.

This only targets the feature branch, so we can always change it if we want again before release.